utils/update-readme.js

```diff
@@ -75,74 +77,37 @@ const kStartMarker = `<!--- Generated by ${path.basename(__filename)} -->`;
 const kEndMarker = `<!--- End of generated section -->`;
 
 /**
- * @param {ParsedToolSchema} tool 
+ * @param {import('../src/tools/tool.js').ToolSchema<any>} tool 
  * @returns {string}
  */
 function formatToolForReadme(tool) {
   const lines = /** @type {string[]} */ ([]);
   lines.push(`<!-- NOTE: This has been generated via ${path.basename(__filename)} -->\n\n`);
-  lines.push(`- **${tool.name}**\n`);
+  lines.push(`- **${tool.title}**\n`);
+  lines.push(`  - Name: \`${tool.name}\`\n`);
```

Comment:
I'd put Title here and keep the tool name in the header as this is an API reference.